src/docfx/docset/Repository.cs

```diff
@@ -46,11 +47,10 @@ public static Repository Create(string path, string branch, string repoUrl = nul
                 return null;
 
             var (remote, repoBranch, commit) = GitUtility.GetRepoInfo(repoPath);
-            var gitIndex = remote.IndexOf(".git");
-            if (gitIndex >= 0)
-            {
-                remote = remote.Remove(gitIndex);
-            }
+
+            // remove user name, token and .git from url like https://xxxxx@dev.azure.com/xxxx.git
+            remote = Regex.Replace(repoUrl ?? remote, @"^((http|https):\/\/)?([^\/\s]+@)?([\S]+?)(\.git)?$", "$1$4");
```

Comment:
I think the conclusion is to not do any url normalization in docfx, no?

Reply:
no, we need do it because docfx is not always used by service. And now the account info will always be part of remote when user clone repo from azure devops:

![image](https://user-images.githubusercontent.com/19990166/57498510-bb435800-730e-11e9-8cee-fed8c12dbda9.png)

Reply:
@live1206 c# `Regex.Replace` also supports `$1`